src/validator/FetchEdgesValidator.cpp

```diff
@@ -139,102 +139,113 @@ Status FetchEdgesValidator::prepareEdges() {
+                return Status::Error("Unknown column `%s' in edge `%s'",
+                                        expr->prop()->c_str(),
+                                        edgeTypeName_.c_str());
             }
-            colNames_.emplace_back(deduceColName(col));
-            auto typeResult = deduceExprType(col->expr());
-            NG_RETURN_IF_ERROR(typeResult);
-            outputs_.emplace_back(colNames_.back(), typeResult.value());
-            // TODO(shylock) think about the push-down expr
+            propsName.emplace_back(*expr->prop());
+            geColNames_.emplace_back(*expr->sym() + "." + *expr->prop());
         }
-        prop.set_props(std::move(propsName));
-    } else {
-        // no yield
-        std::vector<std::string> propNames;   // filter the type
-        propNames.reserve(3 + schema_->getNumFields());
-        outputs_.reserve(3 + schema_->getNumFields());
-        colNames_.reserve(3 + schema_->getNumFields());
-        geColNames_.reserve(3 + schema_->getNumFields());
-        // insert the reserved properties be compatible with 1.0
-        propNames.emplace_back(kSrc);
-        outputs_.emplace_back(edgeTypeName_ + "." + kSrc, Value::Type::STRING);
-        colNames_.emplace_back(edgeTypeName_ + "." + kSrc);
-        geColNames_.emplace_back(colNames_.back());
-        propNames.emplace_back(kDst);
-        outputs_.emplace_back(edgeTypeName_ + "." + kDst, Value::Type::STRING);
-        colNames_.emplace_back(edgeTypeName_ + "." + kDst);
-        geColNames_.emplace_back(colNames_.back());
-        propNames.emplace_back(kRank);
-        outputs_.emplace_back(edgeTypeName_ + "." + kRank, Value::Type::INT);
-        colNames_.emplace_back(edgeTypeName_ + "." + kRank);
-        geColNames_.emplace_back(colNames_.back());
-
-        for (std::size_t i = 0; i < schema_->getNumFields(); ++i) {
-            propNames.emplace_back(schema_->getFieldName(i));
-            outputs_.emplace_back(schema_->getFieldName(i),
-                                  SchemaUtil::propTypeToValueType(schema_->getFieldType(i)));
-            colNames_.emplace_back(edgeTypeName_ + "." + schema_->getFieldName(i));
-            geColNames_.emplace_back(colNames_.back());
-        }
-        prop.set_props(std::move(propNames));
+        colNames_.emplace_back(deduceColName(col));
+        auto typeResult = deduceExprType(col->expr());
+        NG_RETURN_IF_ERROR(typeResult);
+        outputs_.emplace_back(colNames_.back(), typeResult.value());
+        // TODO(shylock) think about the push-down expr
     }
+    prop.set_props(std::move(propsName));
+    props_.emplace_back(std::move(prop));
+    return Status::OK();
+}
+
+Status FetchEdgesValidator::preparePropertiesWithoutYield() {
+    // no yield
+    storage::cpp2::EdgeProp prop;
+    prop.set_type(edgeType_);
+    std::vector<std::string> propNames;   // filter the type
+    propNames.reserve(3 + schema_->getNumFields());
```

Comment:
define `auto size = 3 + schema_->getNumFields()`

Reply:
Not speed more.